pkg/reconciler/certificate/certificate.go

```diff
@@ -184,3 +190,29 @@ func (c *Reconciler) updateStatus(desired *v1alpha1.Certificate) (*v1alpha1.Cert
 
 	return c.ServingClientSet.NetworkingV1alpha1().Certificates(existing.Namespace).UpdateStatus(existing)
 }
+
+func (c *Reconciler) setHTTP01Challenges(knCert *v1alpha1.Certificate) {
+	challenges := []v1alpha1.HTTP01Challenge{}
+	for _, dnsName := range knCert.Spec.DNSNames {
+		domainHash := fmt.Sprintf("%d", adler32.Checksum([]byte(dnsName)))
+		selector := fmt.Sprintf("certmanager.k8s.io/acme-http-domain=%s", domainHash)
+		svcs, _ := c.KubeClientSet.CoreV1().Services(knCert.Namespace).List(metav1.ListOptions{LabelSelector: selector})
+		for _, svc := range svcs.Items {
+			owner := svc.GetOwnerReferences()[0]
+			cmChallenge, _ := c.certManagerClient.CertmanagerV1alpha1().Challenges(knCert.Namespace).Get(owner.Name, metav1.GetOptions{})
+
+			challenge := v1alpha1.HTTP01Challenge{
+				ServiceName:      svc.Name,
+				ServicePort:      intstr.FromInt(8089),
```

Comment:
Is this some magic const?

Reply:
It's the port the cert-manager solver services use. I've changed it to just get the port from the service.